app.rb

```diff
@@ -30,16 +33,8 @@
 
 # let's define a paper route
 get('/paper') do
-  # before we render the paper page, we can add our logic
-  @comp_move = ["rock", "paper", "scissors"].sample
-    
-  if @comp_move == "rock"
-    @outcome = "won"
-  elsif @comp_move == "paper"
-    @outcome =  "tied"
-  elsif @comp_move == "scissors"
-    @outcome = "lost"
-  end
+
+  @game = GameController.new.paper
 
   erb(:paper)#, :layout => :application_layout)
```

Comment:
Could we move the `erb(:paper)` into the "action"? We'd have to include some module, I imagine.

Then I hope the view template wouldn't need to change, and we wouldn't need `@game` at all?

Reply:
@raghubetina This isn't so straightforward without moving the route into the controller (which I guess we definitely don't want?) and/or writing a module that we `include` in the GameController. 

If the intention here is still to make everything explainable to a beginner, I'm not sure it's wise to add more complication that we would (in theory) want to explain?

Here's an example from ChatGPT (very rough cut, but it kind of follows the suggestion in [this SO post](https://stackoverflow.com/questions/2402017/call-sinatra-erb-from-another-class)):

```
require 'sinatra/base'

module Sinatra
  module ControllerRendering
    def render(view, locals = {}, &block)
      template = Tilt.new("#{settings.views}/#{view}.erb")
      template.render(self, locals, &block)
    end
  end

  register ControllerRendering
end

class GameController < Sinatra::Base
  include Sinatra::ControllerRendering

  attr_accessor :comp_move, :outcome

  get('/paper') do
    paper
  end

  def paper
    # move our logic to the game controller
    @comp_move = ["rock", "paper", "scissors"].sample
      
    if @comp_move == "rock"
      @outcome = "won"
    elsif @comp_move == "paper"
      @outcome =  "tied"
    elsif @comp_move == "scissors"
      @outcome = "lost"
    end

    render(:paper, {comp_move: @comp_move, outcome: @outcome})
  end
end
```

Reply:
Hold up, I might have found another way on some more research.

Reply:
I see the `erb` method is defined in `Sinatra::Templates`. Could we include that module in our controller class and then call `erb`?

https://www.rubydoc.info/gems/adamwiggins-sinatra/Sinatra/Templateshttps://www.rubydoc.info/gems/adamwiggins-sinatra/Sinatra/Templates